File: Konwerter_PASCAL_C/PASCAL_files/factorial.pas
program ObliczanieSilni;

{ ==========================================
  1. Wersja rekurencyjna
  ========================================== }
function SilniaRekurencyjna(n: integer): longint;
begin
  if n <= 1 then
    SilniaRekurencyjna := 1
  else
    SilniaRekurencyjna := n * SilniaRekurencyjna(n - 1);
end;

{ ==========================================
  2. Wersja iteracyjna z pętlą REPEAT..UNTIL
  ========================================== }
function SilniaIteracyjna(n: integer): longint;
var
  wynik: longint;
  i: integer;
begin
  wynik := 1;
  i := 1;
  
  if n > 0 then
  begin
    repeat
      wynik := wynik * i;
      i := i + 1;
    until i > n;
  end;
  
  SilniaIteracyjna := wynik;
end;

var
  liczba: integer;

begin
  Write('Podaj liczbe naturalna (>= 0): ');
  ReadLn(liczba);

  if liczba < 0 then
    WriteLn('Blad! Silnia jest zdefiniowana tylko dla liczb nieujemnych.')
  else
  begin
    WriteLn('---');
    WriteLn('Wynik za pomoca rekurencji: ', SilniaRekurencyjna(liczba));
    WriteLn('Wynik za pomoca iteracji:   ', SilniaIteracyjna(liczba));
  end;
  
  WriteLn('Wcisnij ENTER, aby zakonczyc...');
  ReadLn;
end.

File: Konwerter_PASCAL_C/C_files/factorial.c
// Wygenerowano z programu: ObliczanieSilni
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#include <time.h>

int liczba;

int SilniaRekurencyjna(int n) {


    if (n <= 1) {
        return 1;
    } else {
        return n * SilniaRekurencyjna(n - 1);
    }
}

int SilniaIteracyjna(int n) {
    int wynik;
    int i;


    wynik = 1;
    i = 1;
    if (n > 0) {
        do {
            wynik = wynik * i;
            i = i + 1;
        } while (!(i > n));
    }
    return wynik;
}


int main(int argc, char *argv[]) {
    printf("Podaj liczbe naturalna (>= 0): ");
    scanf("%d", &liczba);
    if (liczba < 0) {
        printf("Blad! Silnia jest zdefiniowana tylko dla liczb nieujemnych.\n");
    } else {
        printf("---\n");
        printf("Wynik za pomoca rekurencji: %d\n", SilniaRekurencyjna(liczba));
        printf("Wynik za pomoca iteracji:   %d\n", SilniaIteracyjna(liczba));
    }
    printf("Wcisnij ENTER, aby zakonczyc...\n");
    getchar();
return 0;
}
